client/internal/connect.go

```diff
@@ -68,26 +71,98 @@ func RunClient(ctx context.Context, config *Config) error {
 		if err != nil {
 			return err
 		}
+
+		managementState := nbStatus.ManagementState{
+			URL:       config.ManagementURL.String(),
+			Connected: false,
+		}
+
+		err = statusRecorder.UpdateManagementStatus(managementState)
+		if err != nil {
+			return err
+		}
 		// connect (just a connection, no stream yet) and login to Management Service to get an initial global Wiretrustee config
 		mgmClient, loginResp, err := connectToManagement(engineCtx, config.ManagementURL.Host, myPrivateKey, mgmTlsEnabled,
 			publicSSHKey)
 		if err != nil {
 			log.Debug(err)
-			if s, ok := status.FromError(err); ok && s.Code() == codes.PermissionDenied {
+			if s, ok := gstatus.FromError(err); ok && s.Code() == codes.PermissionDenied {
 				log.Info("peer registration required. Please run `netbird status` for details")
 				state.Set(StatusNeedsLogin)
 				return nil
 			}
 			return wrapErr(err)
 		}
 
+		managementState.Connected = true
+
+		err = statusRecorder.UpdateManagementStatus(managementState)
+		if err != nil {
+			return err
+		}
+
+		localPeerState := nbStatus.LocalPeerState{
+			IP:              loginResp.GetPeerConfig().GetAddress(),
+			PubKey:          myPrivateKey.PublicKey().String(),
+			KernelInterface: iface.WireguardModExists(),
+		}
+
+		err = statusRecorder.UpdateLocalPeerStatus(localPeerState)
+		if err != nil {
+			return err
+		}
+
+		err = statusRecorder.UpdateManagementStatus(managementState)
+		if err != nil {
+			return err
+		}
+
+		signalURL := fmt.Sprintf("%s://%s",
+			strings.ToLower(loginResp.GetWiretrusteeConfig().GetSignal().GetProtocol().String()),
+			loginResp.GetWiretrusteeConfig().GetSignal().GetUri(),
+		)
+
+		signalState := nbStatus.SignalState{
+			URL:       signalURL,
+			Connected: false,
+		}
+
+		err = statusRecorder.UpdateSignalStatus(signalState)
+		if err != nil {
+			return err
+		}
+
+		// set disconnected state for management and signal when exiting
+		defer func() {
+			managementState.Connected = false
+
+			err = statusRecorder.UpdateManagementStatus(managementState)
```

Comment:
I would create a new state and update without messing up with existing objects

Reply:
alternatively create a separate method of statusRecorder that would just update connected state of management, e.g. setManagementConnected(true/false)

Reply:
we can't set the connected status separately as it is related to the URL